src/frontend/src/handler/create_sink.rs

```diff
@@ -533,6 +533,14 @@ fn check_cycle_for_sink(
             }
         }
 
+        for table_id in &table.dependent_relations {
```

Comment:
Can you explain in detail what issues this PR fixed? Is it here to check whether there is a cycle to the other join paths of the streaming job?